packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -764,6 +771,31 @@ export class Combobox
     this.el.removeEventListener("calciteComboboxOpen", this.toggleOpenEnd);
   };
 
+  private setTooltipText = (): void => {
+    const { textLabelEl } = this;
+    if (!textLabelEl) {
+      return;
+    }
+
+    requestAnimationFrame(() => {
+      this.tooltipText =
+        textLabelEl.offsetWidth < textLabelEl.scrollWidth ? textLabelEl.innerText : null;
```

Comment:
I would compare the actual text width using [this util](https://github.com/Esri/calcite-design-system/blob/34896cbd6eb2600823446f5f0e7b17bb2d42ba25/packages/calcite-components/src/utils/dom.ts#L126-L134) with the computed width of the text's container.  Also, rather than making a separate `@State` property for the tooltip text, just use the selected item's display value and in the render method, just conditionally render the `title` attribute.

Reply:
I don't think that util makes sense in this scenario. We just want to know if the offsetWidth is less than the scrollWidth which is simpler than creating a canvas and measuring.

I do agree that the `@State()`isn't ideal. It could probably just be a boolean for whether to render the title or not and use the selected items display value. I think it still needs a state because the render method wouldn't update when the resize observer fires unless it tells some state to change.

Reply:
Note: Button seems to do the same logic so a util that handles this might be useful. Something like `hasScrollbar(el);`

https://github.com/Esri/calcite-design-system/blob/210e2da30946664a803bdb2cb67d2e83b6fc7532/packages/calcite-components/src/components/button/button.tsx#L414-L420

Reply:
Yeah I thought that too. IMO that would be the simplest option. It is a tiny bit odd to conditionally render a title attribute.

Reply:
> I also thought, why not just always render the full `title` attribute? Is the intention to not unnecessarily crowd the UI when hovering the selected item with the mouse if the full text is visible? Either way, rendering the title all the time would just be the simplest approach, no need to calculate overflow whatsoever... Just some food for thought.

@ashetland what do you think of just always rendering the `title` attribute?

Reply:
Honestly not sure why this hasn't come up before 😆. I don't see any problems with that. I think there are a few other places we're using the `title` attribute like this, but I'm not sure if any of them are conditional. Seems like we might be able to always render `title` everywhere we do this.

Reply:
This has come up before in the button truncation. We had a full discussion with @Aaron and @SkyeSeitz about it and there was a small visual to illustrate the crowding. The gist of it was:

To avoid unnecessary noise, detect the buttons with text that has been truncated and apply title attribute conditionally only to these, to reveal the full text for sighted users. This frees up the calcite-tooltip to provide additional context.

https://github.com/Esri/calcite-design-system/pull/6664#issuecomment-1485643872

Reply:
Button was the only other place we seem to be doing this conditional logic. It would be much simpler and easier on performance if we just always set the title attribute when we are potentially truncating.

Reply:
I think this might have come up with tab-title truncation as well.

Reply:
Thanks @Elijbet for the refresher! If we play this out to possible truncation behavior in places like List or Tree, I agree that keeping this conditional is a cleaner UX for the end user. If the text is short enough to be displayed then there isn't much reason to also render the `title` attribute. Add a possible `calcite-tooltip` on top of that and there's suddenly a lot going on. Reducing redundancy feels like a win.

Reply:
We've also been talking about mapping out a system-wide configurable `text-overflow`. With that in mind and since, in this case, the entire text can be read by expanding the Combobox, I want to put it out there that we could abandon this for the time being.

Reply:
Ok, that makes sense to me.

Reply:
Closing the issue then.